### DISCUSSION TITLE
Battery Temperature available from Tesla API?

### DISCUSSION THREAD

**Discussion by fhteagle** [alandtse/tesla#549] [General]
Hi all, and thanks for this very useful integration.

I have defined automations to start the preconditioning process (and set my OpenEVSE to active) at a time that varies based on cabin and outside temperature, and so far these are working well. However,  I have noticed that the time it takes for battery preconditioning is not strongly correlated to either of the above, and sometimes is not fully done by actual departure time, or completes quite too soon. I could continue to fiddle with my automation logic, or try to find a more accurate piece of data to calculate from. 

I looked through third party descriptions of the Tesla API, and did not see a current battery temperature data point reported. Were they missed in the API writeup, do they not get reported at all by Tesla, or could they be easily added to the HA integration? If the latter, please consider this a feature request for the same.

Thanks again, and happy EV'ing...

**Comment by Joe123-B** [comment]
was looking for the same info; Battery temperature - if it's a data that can be reported through this integration, everything else is there.

also have you managed to start the battery preconditioning process manually?
